### DISCUSSION TITLE
_why_ does the default selection key `toggle; down`? should we keep this behavior?

### DISCUSSION THREAD

**Discussion by sudoforge** [gokcehan/lf#1297] [General]
for context, i mostly use `lf` as a filepicker; i don't tend to have long-lived sessions interacting with my filesystem.

if i'm selecting a single file, i'll usually navigate to it and open it (to submit my selection to the website). if i'm selecting multiple files, i'll navigate to each of them, using `<space>` to select each as i go. then i get to the final one, toggle it, and then need to move `up` before "opening" to submit my selection. i find the need to move back `up` to be counterintuitive.

the only conversation i could find about _why_ we `toggle ; down` is in #352, in which it seems like `:toggle` used to contain its own logic to move down, this was considered a bug, so we changed the implementation but re-mapped `<space>` from `:toggle` to `:toggle;down`.

this doesn't provide a whole lot of context as to _why_ we move down after selection, other than "we should do this to keep the default behavior". i find that this default behavior is never what i actually want to do, adding an extra step (`:up`) prior to opening my file(s).

**Comment by joelim-work** [comment]
The default `toggle; down` behaviour is useful when you want to select some files while navigating down a directory. When you toggle a file, you're probably not interested in toggling that file again, so the cursor advances to the next file as a convenience. This results in a workflow where you navigate downwards, using `<space>` to toggle a file, or `j` to skip past it.

As for opening files, the default `open` command opens `$f`, which is the current file located at the cursor (the selection list is not considered here). The reason for this is to prevent users from accidentally selecting multiple files of different types and trying to open them all with the same program. Since `open` only considers the current file located at the cursor, you don't need to toggle with `<space>` at all.

---

BTW I'm actually not 100% sure of your workflow - from your other post https://github.com/gokcehan/lf/discussions/1080#discussioncomment-6139735, it appears you're using the `-selection-path` mode? In this case the selection list is the final result, and the current file located at the cursor is not considered, in which case moving back `up` is not necessary.

**Reply by sudoforge** [reply]
i use both the "normal" mode (irregularly) and `--selection-path` (far more regularly). perhaps i'm not the typical user, but i very rarely, if ever, process a directory, acting on each file in succession -- i'm much more likely to be picking a variety of files, all over the place, moving to each of them with `<#>j/k` or via searching.

> BTW I'm actually not 100% sure of your workflow - from your other post #1080 (comment), it appears you're using the -selection-path mode? In this case the selection list is the final result, and the current file located at the cursor is not considered, in which case moving back up is not necessary.

neat! i was not aware of this; thanks for the heads up.
